### ISSUE TITLE
deepcopy of SimpleVector corrupts memory

### ISSUE THREAD

**Issue by mauro3** [JuliaLang/julia#15250]
This bug happens on 0.4.3
```
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.3 (2016-01-12 21:37 UTC)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |  x86_64-unknown-linux-gnu

julia> let
       a = 5
       global an = ()->a
       end
(anonymous function)

julia> ann = deepcopy(an); ann()
Base.KeyIterator
```
The let block is necessary.

It works fine on 0.5. (I don't need this fixed on 0.4.3, but I thought I report it.  Please close if won't-fix.)

**Comment by yuyichao** [comment]
I think this is a won't fix on 0.4. We could add test to make sure it works on 0.5+ though.

**Comment by vtjnash** [comment]
simple repro:
```
julia> Base.svec(1,2,3)
svec(1,2,3)

julia> Base.deepcopy(ans)
svec(Tuple{SimpleVector,Void},#= circular reference =#,Box)
```
